File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        Minprice = prices[0]
        profit = 0
        maxp = 0
        n = len(prices)
        for i in range(0,n):
            Minprice = min(Minprice,prices[i])
            profit = prices[i] - Minprice
            maxp = max(profit,maxp)
        return maxp

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
      int minprice = Integer.MAX_VALUE;
      int maxprice = 0;

        for(int price:prices)
        {
            minprice = Math.min(price,minprice);

            int profit = price - minprice;

            maxprice = Math.max(profit,maxprice);
        }

        return maxprice;
    }
}